lib/getstats.js

```diff
@@ -316,6 +320,41 @@ function chromeGetTrackStats(peerConnection, track) {
   });
 }
 
+/**
+ * Get the standardized statistics for a particular MediaStreamTrack in Edge.
+ * @param {RTCPeerConnection} peerConnection
+ * @param {MediaStreamTrack} track
+ * @returns {Promise.<StandardizedTrackStatsReport>}
+ */
+function edgeGetTrackStats(peerConnection, track) {
+  return new Promise(function(resolve, reject) {
+    waitForEdgeIceCompletion(peerConnection).then(function() {
+      peerConnection.getStats(track).then(function(response) {
+        resolve(standardizeFirefoxStats(response));
+      }, reject);
+    });
+  });
+}
+
+function waitForEdgeIceCompletion(peerConnection) {
```

Comment:
@syerrapragada Since there is nothing `edge`-specific in the way you've used the RTCPeerConnection API here, you can refactor this method like so:
```js
function waitForIceCompletion(peerConnection) {
  const iceConnectionStatesToWait = new Set(['checking', 'connected', 'new']);
  if (!iceConnectionStatesToWait.has(peerConnection)) {
    return Promise.resolve();
  }
  return new Promise(function(resolve, reject) {
    const timeoutToFail = setTimeout(reject, FAIL_DELAY, new Error('failed'));
    peerConnection.addEventListener('iceconnectionstatechange', function onIceConnectionStateChanged() {
        const iceConnectionStatesToResolve = new Set(['completed', 'failed']);
        if (iceConnectionStatesToResolve.has(peerConnection.iceConnectionState)) {
          peerConnection.removeEventListener('iceconnectionstatechange, onIceConnectionStateChanged);
          setTimeout(resolve, 4000);
        }
    });
  });
}
```